packages/cells/src/cells/date-picker-cell.tsx

```diff
@@ -18,46 +101,20 @@ const renderer: CustomRenderer<DatePickerCell> = {
         drawTextCell(args, displayDate, cell.contentAlign);
         return true;
     },
-    // eslint-disable-next-line react/display-name
-    provideEditor: () => p => {
-        const cellData = p.value.data;
-        const { format, date } = cellData;
-
-        let val = "";
-        if (date !== undefined) {
-            val = date.toISOString();
-            if (format === "date") {
-                val = val.split("T")[0];
-            } else {
-                val = val.substring(0, 23);
-            }
-        }
-        return (
-            <input
-                style={{ minHeight: 26, border: "none", outline: "none" }}
-                type={format}
-                autoFocus={true}
-                value={val}
-                onChange={e => {
-                    p.onChange({
-                        ...p.value,
-                        data: {
-                            ...p.value.data,
-                            date: e.target.valueAsDate ?? undefined,
-                        },
-                    });
-                }}
-            />
-        );
-    },
+    provideEditor: () => ({
+        editor: Editor,
+    }),
     onPaste: (v, d) => {
         let newDate: Date | undefined;
         try {
             newDate = new Date(v);
         } catch {
             /* do nothing */
         }
-
+        if (Number.isNaN(newDate)) {
+            // Try to interpret value as time string (HH:mm:ss)
+            newDate = new Date(`1970-01-01T${v}Z`);
+        }
         return {
             ...d,
             date: Number.isNaN(newDate) ? undefined : newDate,
```

Comment:
Oh, I think `Number.isNaN` is actually not working correctly here. For example,  `new Date("foo")` will not be detected via `Number.isNaN`. This only works in combination with `Date.parse`. There are a few information [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/parse#fall-back_to_implementation-specific_date_formats).

Reply:
I will add a fix

Reply:
I think this could be `Number.isNaN(newDate.getTime())`

Reply:
yep, but I think it's better to directly use `Date.parse`, since this is anyways used when called via `new Date()`